.github/workflows/miden.yml

```diff
@@ -0,0 +1,124 @@
+    runs-on: ubuntu-latest
+    needs: setup-ec2
+    steps:
+      - name: Check for updates on crates.io
+        run: |
+          NEW_VERSION=$(curl -s https://crates.io/api/v1/crates/miden-node | jq -r '.crate.max_version')
+          CURRENT_VERSION=$(ssh -i miden-test.pem -o "StrictHostKeyChecking no" ec2-user@54.154.193.162 "miden-node --version | cut -d ' ' -f2")
+          if [ "$NEW_VERSION" != "$CURRENT_VERSION" ]; then
+            echo "::set-output name=update_needed::true"
+          else
+            echo "::set-output name=update_needed::false"
+          fi
+
+  update-ec2:
+    runs-on: ubuntu-latest
+    needs: check-crates
+    if: needs.check-crates.outputs.update_needed == 'true'
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::211125515935:role/Miden-GithubActionsRole
+          aws-region: us-west-2
+
+      - name: SSH into EC2 and upgrade miden-node
+        run: |
+          ssh -i miden-test.pem -o "StrictHostKeyChecking no" ec2-user@54.154.193.162 << 'EOF'
+            # Stop the miden-node service
+            sudo systemctl stop miden-node
+            sudo systemctl stop miden-faucet
+
+            # Remove previous data
+            # rm -rf /home/ec2-user/miden-node
+            # rm -rf /home/ec2-user/miden-faucet
+
+            rm -rf ~/miden-node; mkdir ~/miden-node
+            rm -rf ~/miden-faucet; mkdir ~/miden-faucet
+
+            # Install latest versions of miden-node and miden-faucet
+            cargo install miden-node miden-faucet --locked --features testing
+
+            # Initialize the services again
+            # miden-node init
+            # miden-faucet init
+
+            miden-node init -c ~/miden-node/miden-node.toml -g ~/miden-node/genesis.toml
+            miden-node make-genesis -i ~/miden-node/genesis.toml -o ~/miden-node/genesis.dat
+            miden-faucet init -c ~/miden-faucet/miden-faucet.toml
+
+            # Configure systemd service
+            sudo tee /etc/systemd/system/miden-node.service > /dev/null <<EOT
+            [Unit]
+            Description=Miden Node
+            After=network.target
+
+            [Service]
+            ExecStart=/home/ec2-user/.cargo/bin/miden-node start -c miden-node.toml node
+            ExecStart=/home/ec2-user/.cargo/bin/miden-faucet start -c miden-faucet.tomls
+            Restart=on-failure
+            User=ec2-user
+            WorkingDirectory=/home/ec2-user/miden-node
+            WorkingDirectory=/home/ec2-user/miden-faucet
+
+            Environment="RUST_LOG=info"
+            LimitNOFILE=10000
+
+            [Install]
+            WantedBy=multi-user.target
+            EOT
+
+            sudo systemctl daemon-reload
+            sudo systemctl enable miden-node
+            sudo systemctl start miden-node
+            sudo systemctl enable miden-faucet
+            sudo systemctl start miden-faucet
```

Comment:
@bobbinth @phklive do we need a delay between starting the node and the faucet?

Reply:
We need to make sure that the node is up and running before starting the faucet considering that the faucet will try to connect to it. Hence yes we could add a delay or find another solution to make sure that the node is up before attempting faucet connection.